### PR TITLE
Fix false negatives for `invalidOptionWarnings`

### LINKED ISSUE #58: Assert plugin options validation
### What is the problem you're trying to solve?

I want to assert my plugins options validation.

[`stylelint-test-rule-tape`](https://github.com/stylelint/stylelint-test-rule-tape) allowed to `reject` for invalid options.

This doesn't work anymore with jest.

### What solution would you like to see?

The same behavior of invalid plugin options with `reject` as it was working in [`stylelint-test-rule-tape`](https://github.com/stylelint/stylelint-test-rule-tape).

The resolved promise of the `lint()` API holds that data in [`invalidOptionWarnings`](https://github.com/stylelint/stylelint/search?q=invalidOptionWarnings).

### PR DESCRIPTION
**Pull request by AndyOGo**
> Which issue, if any, is this issue related to?

Fixes #58 

> Is there anything in the PR that needs further explanation?

No, it's self-explanatory.

### CODE REVIEW

**Comment by ybiquitous** [comment]
@AndyOGo Thanks for the pull request. Just in case, I confirmed this change for the `stylelint/stylelint` repo.

1. Edit `package.json` to install the version of this pull request:

```diff
--- a/package.json
+++ b/package.json
@@ -182,7 +182,7 @@
     "eslint-plugin-jest": "^27.2.1",
     "husky": "^8.0.3",
     "jest": "^29.4.2",
-    "jest-preset-stylelint": "^6.0.0",
+    "jest-preset-stylelint": "github:stylelint/jest-preset-stylelint#pull\/59\/merge",
     "jest-watch-typeahead": "^2.2.2",
     "lint-staged": "^13.1.1",
     "node-fetch": "^3.3.0",
```

2. Run `npm install`
3. Run `npm test`

Then, I got the test failures like this:

```
  ● selector-max-combinators › accept › [ true ] › '.while(@n: 0) when (@n < 10) { .n-@{n} { content: %("n: %d", 1 + 1); } .while(@n + 1) }' › ignore Less interpolation inside .while

    expect(received).toEqual(expected) // deep equality

    - Expected  - 1
    + Received  + 5

    - Array []
    + Array [
    +   Object {
    +     "text": "Invalid option \"true\" for rule \"selector-max-combinators\"",
    +   },
    + ]
```

This is because [the `selector-max-combinators` test case specifies a wrong config value: `true`](https://github.com/stylelint/stylelint/blob/063d25d91227d2cd38980677fb98201761f1e75b/lib/rules/selector-max-combinators/__tests__/index.js#L361). Correctly, we need to specify a number value instead of `true`.

In short, this pull request can detect our test bug! Thank you.

**Review by ybiquitous — APPROVED** [review]
LGTM 👍🏼

**Review by jeddy3 — APPROVED** [review]
Fantastic, thank you!

**Comment by jeddy3** [comment]
Change entry added:

> - Fixed: false negatives for `invalidOptionWarnings`.
